app/src/main/java/com/yacgroup/yacguide/network/RockParser.kt

```diff
@@ -103,26 +104,28 @@ class RockParser(
             r.firstAscendDate = jsonRoute.getString("erstbegdatum")
             r.typeOfClimbing = jsonRoute.getString("kletterei")
             r.description = ParserUtils.jsonField2String(jsonRoute, "wegbeschr_d", "wegbeschr_cz")
-            val ascends = db.ascendDao().getAscendsForRoute(routeId) // if we re-add a route that has already been marked as ascended in the past
-            for (ascend in ascends) {
-                r.ascendsBitMask = (r.ascendsBitMask or AscendStyle.bitMask(ascend.styleId))
-            }
+            // if we re-add a route that has already been marked as ascended in the past:
+            _db.getRouteAscends(routeId).map { r.ascendsBitMask = (r.ascendsBitMask or AscendStyle.bitMask(it.styleId)) }
             r.parentId = ParserUtils.jsonField2Int(jsonRoute, "gipfelid")
-            db.routeDao().insert(r)
+            routes.add(r)
         }
+        _db.refreshRoutes(routes, _sectorId)
 
         // update already ascended rocks
-        for (rock in db.rockDao().getAll(_sectorId)) {
-            for (ascend in db.ascendDao().getAscendsForRock(rock.id)) {
-                rock.ascendsBitMask = (rock.ascendsBitMask or AscendStyle.bitMask(ascend.styleId))
+        val updatedRocks = _db.getRocks(_sectorId).map {
+            for (ascend in _db.getRockAscends(it.id)) {
+                it.ascendsBitMask = it.ascendsBitMask or AscendStyle.bitMask(ascend.styleId)
             }
-            db.rockDao().setAscendsBitMask(rock.ascendsBitMask, rock.id)
+            it
```

Comment:
What is that?

Reply:
Kotlin-specific handler of the current element in the loop